tools/bin/check_images_exist.sh

```diff
@@ -5,9 +5,20 @@ set -e
 . tools/lib/lib.sh
 
 function docker_tag_exists() {
+  echo $1
+  if [[ $1 == ghcr* ]]
+  then
+    # As of right now ghcr.io API is still under progress and it doesn't provide 
+    # manifest API similar to docker so added this check
+    IMAGE_WITH_VERSION="$1":"$2"
+    printf "\tIMAGE WITH VERSION: %s\n" "$IMAGE_WITH_VERSION"
+    docker pull $IMAGE_WITH_VERSION
```

Comment:
is there a url that we can curl instead of actually pulling the image down? the approach for dockerhub is to just to a get request to see that the image exists. does ghcr do something similar? if so we should prefer it.

Reply:
Looks like it is possible: https://github.community/t/how-to-check-if-a-container-image-exists-on-ghcr/154836/4

Reply:
Yeah, you would need token and authentication to do that. Since the number of images are less can we do that once they bring up a public API similar to docker

Or is it that we want to add token and use it? What do you folks suggest?

Reply:
The post mentioned that if it is a public image, a fake token is sufficient. We are dealing with public images, right? If that's the case, I think it's worth trying it.

Otherwise, I think it is fine to pull the image then. Feel free to merge it.

Reply:
Makes sense. Let me try it with fake token